storagetransfer/event_driven_gcs_transfer.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetransfer
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	storagetransfer "cloud.google.com/go/storagetransfer/apiv1"
+	"cloud.google.com/go/storagetransfer/apiv1/storagetransferpb"
+)
+
+// [START storagetransfer_create_event_driven_gcs_transfer]
+
+func createEventDrivenGcsTransfer(w io.Writer, projectID string, gcsSourceBucket string, gcsSinkBucket string, pubSubId string) (*storagetransferpb.TransferJob, error) {
+	// Your Google Cloud Project ID.
+	// projectID := "my-project-id"
+
+	// The name of the source GCS bucket.
+	// gcsSourceBucket := "my-source-bucket"
+
+	// The name of the GCS bucket to transfer objects to.
+	// gcsSinkBucket := "my-sink-bucket"
+
+	// The Pub/Sub topic to subscribe the event driven transfer to.
+	// pubSubId := "projects/PROJECT_NAME/subscriptions/SUBSCRIPTION_ID"
```

Comment:
Could you just pass in the subscription ID and build this using the project ID and subscription ID?

Reply:
I could,  but given that the API gives back this whole thing when you query for it I figured it would be the easier thing to do in most cases